js/tuxexchange.js

```diff
@@ -0,0 +1,567 @@
+                // precision: precision, // TODO: not listed? infer?
+                'info': tickerData,
+                // limits: limits,
+            };
+            result.push (market);
+        }
+        return result;
+    }
+
+    async fetchCurrencies (params = {}) {
+        const res = await this.publicGet ({ 'method': 'getcoins' });
+        const currenciesData = this.parseJson (res);
+        const currencyIds = Object.keys (currenciesData);
+        // The API does not expose BTC as a coin (I suspect because it is the base in each market)
+        let result = { 'BTC': {
+            'id': 'BTC',
+            'code': 'BTC',
+            'name': 'bitcoin',
+            'fiat': false,
+        }};
+        for (let i = 0; i < currencyIds.length; i++) {
+            const id = currencyIds[i];
+            const coinData = currenciesData[id];
+            const code = this.commonCurrencyCode (id);
+            result[code] = {
+                'id': id,
+                'code': code,
+                'name': coinData['name'],
+                'fiat': false,
+                'funding': {
+                    'withdraw': {
+                        'fee': this.safeFloat (coinData, 'withdrawfee'),
+                    },
+                },
+            };
+        }
+        return result;
+    }
+
+    fetchTradingFees (params = {}) {
+        return {
+            'maker': 0,
+            'taker': 0.03,
+        };
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchTicker () requires a "symbol" argument');
+        }
+        let res = await this.publicGet ({ 'method': 'getticker' });
+        const tickersData = this.parseJson (res);
+        const tickerIds = Object.keys (tickersData);
+        for (let i = 0; i < tickerIds.length; i++) {
+            const id = tickerIds[i];
+            const tickerResult = this.parseTicker (id, tickersData[id]);
+            const tickerSymbol = tickerResult['symbol'];
+            if (tickerSymbol === symbol) {
+                return tickerResult;
+            }
+        }
+        throw new BadRequest (this.id + ' no ticker found for symbol provided');
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        await this.loadMarkets ();
+        let res = await this.publicGet ({ 'method': 'getticker' });
+        const tickersData = this.parseJson (res);
+        const tickerIds = Object.keys (tickersData);
+        const result = {};
+        for (let i = 0; i < tickerIds.length; i++) {
+            const id = tickerIds[i];
+            const tickerResult = this.parseTicker (id, tickersData[id]);
+            const symbol = tickerResult['symbol'];
+            result[symbol] = tickerResult;
+        }
+        return result;
+    }
+
+    parseTicker (marketId, tickerData) {
+        const timestamp = this.nonce (); // TODO: hopefully this approximation is sufficient
```

Comment:
This can cause some issues -- the implementation for `nonce` can get replaced by library users, so you're best off just using `now ()` or `milliseconds ()`

Reply:
done